File: domain/Validators/ArtistValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LabMPP.domain.Validators
{
    class ArtistValidator : Validator<Artist>
    {
        public void validate(Artist entity)
        {
            if (entity.name=="" || entity.genre=="")
                throw new ValidationException("Numele si genreul nu pot fi vide!");
        }
    }
}

File: Model/src/main/java/domain/validators/ArtistValidator.java
package domain.validators;

import domain.Artist;

import java.io.Serializable;

public class ArtistValidator implements Validator<Artist> , Serializable {
    @Override
    public void validate(Artist entity) throws ValidationException {
        if(entity.getName().isEmpty() || entity.getGenre().isEmpty())
            throw new ValidationException("Numele si genreul nu pot fi vide!");
    }
}
